sdlib/d/gc/arena.d

```diff
@@ -148,6 +148,55 @@ public:
 		return null;
 	}
 
+	bool resizeLarge(shared(ExtentMap)* emap, ref PageDescriptor pd,
+	                 size_t size) shared {
+		assert(pd.extent !is null, "Extent is null!");
+		assert(pd.extent.arenaIndex == index, "Invalid arena index!");
+		assert(isLargeSize(size));
+
+		if (pd.isSlab()) {
+			return false;
+		}
+
+		auto computedPageCount = alignUp(size, PageSize) / PageSize;
+		uint newPages = computedPageCount & uint.max;
+
+		assert(newPages == computedPageCount, "Unexpected page count!");
+
+		auto e = pd.extent;
```

Comment:
It doesn't look like pd is used for anything else, so why not take the extent as argument?